lib/presentation/widgets/list_characters.dart

```diff
@@ -18,18 +22,31 @@ class ListCharacters extends StatelessWidget {
         }
         if (state.loading == false && state.charactersViewData != null) {
           return ListView.builder(
+            controller: _scrollController
+              ..addListener(
```

Comment:
Almost every time you adding some listener - you have to remove it when is not needed, for instance when the page is closed. Also _scrollController should be disposed at some point.